pkg/controller/controller.go

```diff
@@ -184,6 +186,27 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 		}
 	}
 
+	func() {
+		waitGroup.Add(1)
+		go func() {
+			wait.Until(func() {
+				glog.V(9).Info("cluster ID monitor loop enter")
+				_, err := c.serviceCatalogClient.ClusterIDs().Get("cluster-id", metav1.GetOptions{})
```

Comment:
It is overkill to query every second for whether the `ClusterID` exists. Create a controller to handle re-creating the `ClusterID` when it is deleted and use a watch of the `ClusterID` resources.

Reply:
No idea which way is better, this this is consistent with what they do for namespaces:  https://github.com/kubernetes/kubernetes/blob/master/pkg/master/controller.go#L146

Reply:
Fair enough.

Reply:
That chunk of code looks to be very old and original. I think there have been better patterns created like the one @staebler suggests.